ktlint-ruleset-experimental/src/main/kotlin/com/pinterest/ktlint/ruleset/experimental/MultiLineIfElseRule.kt

```diff
@@ -22,14 +28,33 @@ class MultiLineIfElseRule : Rule("multiline-if-else") {
             if (!node.treePrev.textContains('\n')) { // if (...) <statement>
                 return
             }
+
             if (node.firstChildNode?.firstChildNode?.elementType != LBRACE) {
                 emit(node.firstChildNode.startOffset, "Missing { ... }", true)
-                // Disabled until we autocorrect with proper formatting
-                /*if (autoCorrect) {
-                    (node.firstChildNode.firstChildNode as TreeElement).rawInsertBeforeMe(LeafPsiElement(RBRACE, "{"))
-                    (node.lastChildNode.lastChildNode as TreeElement).rawInsertAfterMe(LeafPsiElement(LBRACE, "}"))
-                }*/
+                if (autoCorrect) {
+                    autocorrect(node)
+                }
             }
         }
     }
+
+    private fun autocorrect(node: ASTNode) {
+        val bodyIndent = node.treePrev.text
```

Comment:
I think this still won't cover all the use cases.. so if you have a `if (true) return 0` on the same line, than it should be
```
if (true) {
    return 0
}
```
but this will format it to
```
if (true) {
 return 0
}
```

I guess you better take a look at the [IndentationRule](https://github.com/pinterest/ktlint/blob/master/ktlint-ruleset-experimental/src/main/kotlin/com/pinterest/ktlint/ruleset/experimental/IndentationRule.kt) and try to replicate its behaviour in regards to if/else indentation.

Or, I'm wondering, if it's possible to just call `IndentationRule().format(KtBlockExpression.text)` ? Never seen that the rules can invoke other rules though.. perhaps, it would be more applicable to extract that logic from the IndentationRule to some common place, so other rules can reuse that for indentation when needed.. what are your thoughts @shashachu @Tapchicoma ?

Reply:
This rule allows single line if statements so it would not change `if (true) return 0`. One of the test cases verifies this. Are there other cases where the resulting indentation would be wrong?

Reply:
True, sorry, I totally missed that 🤦 Can you maybe add one more test just to make sure (although it shouldn't change anything), with multiline condition within `if` block? Like
```
        if (i2 > 0 &&
            i3 < 0
        ) {
            return 2
        }
```